metals/src/main/scala/scala/meta/internal/metals/DefinitionProvider.scala

```diff
@@ -508,3 +537,96 @@ class DestinationProvider(
     }
   }
 }
+
+class DefinitionProviderReportBuilder(
+    path: AbsolutePath,
+    params: TextDocumentPositionParams,
+) {
+  private var compilerDefn: Option[DefinitionResult] = None
+  private var semanticDBDefn: Option[DefinitionResult] = None
+
+  private var fallbackDefn: Option[DefinitionResult] = None
+  private var nonLocalGuesses: List[String] = List.empty
+
+  private var fundScaladocDef = false
+
+  private var error: Option[Throwable] = None
+
+  def withCompilerResult(result: DefinitionResult): DefinitionResult = {
+    compilerDefn = Some(result)
+    result
+  }
+
+  def withSemanticDBResult(result: DefinitionResult): DefinitionResult = {
+    semanticDBDefn = Some(result)
+    result
+  }
+
+  def withFallbackResult(result: DefinitionResult): DefinitionResult = {
+    fallbackDefn = Some(result)
+    result
+  }
+
+  def withError(e: Throwable): Unit = {
+    error = Some(e)
+  }
+
+  def withNonLocalGuesses(guesses: List[String]): Unit = {
```

Comment:
```suggestion
  def setNonLocalGuesses(guesses: List[String]): Unit = {
```
This is more a setter than a builder method, no? Could we make them more consistent or rename as above?